### PR TITLE
Support processing of uploaded zip archives

### PR DESCRIPTION
**Pull request by kgevorkyan**
### What's done:
* Support processing of uploaded zip archives - unzip zip archives into docker image among provided additional files

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -6,6 +6,7 @@ okio = "3.0.0-alpha.10"
 serialization = "1.3.1"
 kotlinx-datetime = "0.3.1"
 kotlinx-coroutines = "1.5.2"
+kotlin-zip4j = "2.9.1"
```
Comment (petertrr) [inline gradle/libs.versions.toml:9]:
```suggestion
zip4j = "2.9.1"
```
This is pure Java library? Why prefix it with `kotlin-`?

File: save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/service/DockerService.kt [hunk 2]
```diff
@@ -240,6 +254,63 @@ class DockerService(private val configProperties: ConfigProperties) {
...
+
+    @Suppress("MAGIC_NUMBER", "LONG_NUMERICAL_VALUES_SEPARATED", "MagicNumber")
+    private fun File.isValidZipArchive(): Boolean {
+        var fileSignature: Int
+        try {
+            RandomAccessFile(this, "r").use { raf -> fileSignature = raf.readInt() }
+        } catch (e: IOException) {
+            log.error("Error during checking file ${this.name} for being an archive: ${e.message}")
+            fileSignature = 0
+        }
```
Comment (petertrr) [inline save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/service/DockerService.kt:300]:
```suggestion
        val fileSignature: Int = try {
            RandomAccessFile(this, "r").use { raf -> raf.readInt() }
        } catch (e: IOException) {
            log.error("Error during checking file ${this.name} for being an archive: ${e.message}")
            0
        }
```

File: save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/service/DockerService.kt [hunk 3]
```diff
@@ -240,6 +254,63 @@ class DockerService(private val configProperties: ConfigProperties) {
...
+        // check for 'magic numbers', i.e. the data used to identify or verify the content of a file
+        return fileSignature == 0x504B0304 || fileSignature == 0x504B0506 || fileSignature == 0x504B0708
```
Comment (petertrr) [inline save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/service/DockerService.kt:302]: Doesn't the library handle such checks by itself? Why is this check needed?

File: gradle/libs.versions.toml (hunk 1 above)
Comment (kgevorkyan) [reply gradle/libs.versions.toml]: Fixed

File: save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/service/DockerService.kt (hunk 2 above)
Comment (kgevorkyan) [reply save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/service/DockerService.kt]: Removed this method

File: save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/service/DockerService.kt (hunk 3 above)
Comment (kgevorkyan) [reply save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/service/DockerService.kt]: Yeah, let it be covered by library

**Review by petertrr — APPROVED** [review]

**Pull request merged**